numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -2098,6 +2098,17 @@ array_fromstring(PyObject *NPY_UNUSED(ignored), PyObject *args, PyObject *keywds
         Py_XDECREF(descr);
         return NULL;
     }
+
+    // binary mode, condition copied from PyArray_FromString
```

Comment:
C++ comments not allowed, use `/* ... */`. This might change when we eventually require C99, but due to windows that will not happen until we drop Python 2.7, 3.4.

I'll fix this and merge.